infra/relayer/src/purse.js

```diff
@@ -0,0 +1,696 @@
+/**
+ * Wallet abstraction for high-volume transactions
+ *
+ * References
+ * ----------
+ * BIP44 - HD Wallet paths - https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
+ * Ethereum and BIP44 discussion - https://github.com/ethereum/EIPs/issues/84
+ *
+ * TODO
+ * ----
+ * - Locks and support for parallelism so multiple instances can work at the same time?
+ * - Auto-scale children when all accounts hit the max pending setting?
+ * - If the service is restarted, tracking and callbacks are lost for pending transactions.  How to deal?
+ */
+const { promisify } = require('util')
+const redis = require('redis')
+const BN = require('bn.js')
+const bip39 = require('bip39')
+const hdkey = require('ethereumjs-wallet/hdkey')
+const {
+  stringToBN,
+  numberToBN,
+  getBIP44Path,
+  bufferToHex,
+  sendRawTransaction
+} = require('./util')
+const logger = require('./logger')
+
+const REDIS_RETRY_TIMEOUT = 30000
+const REDIS_RETRY_DELAY = 500
+const DEFAULT_CHILDREN = 5
+const DEFAULT_MAX_PENDING_PER_ACCOUNT = 3
+const ZERO = new BN('0', 10)
+const BASE_FUND_VALUE = new BN('50000000000000000', 10) // 0.05 Ether
+const MIN_CHILD_BALANCE = new BN('1000000000000000', 10) // 0.001 Ether
+const MAX_GAS_PRICE = new BN('20000000000', 10) // 20 gwei
+const REDIS_TX_COUNT_PREFIX = 'txcount_'
+
+async function tick(wait = 1000) {
+  return new Promise(resolve => setTimeout(() => resolve(true), wait))
+}
+
+class Account {
+  constructor({ txCount, pendingCount = 0, wallet, funded }) {
+    this.txCount = txCount
+    this.pendingCount = pendingCount // needed?
+    this.wallet = wallet
+    this.funded = funded
```

Comment:
I'm not super clear on the role of the funded flag in the constructor here as well as in the Account objects. It does not seem to be used for any logic ? Shall it be removed then ?

Reply:
Appears to be an artifact.  I was using it at one point, but no longer.  Will remove, thanks for catching that.